### ISSUE TITLE
handle etext access control

### ISSUE THREAD

**Issue by eroux** [buda-base/public-digital-library#916]
I finished the work on the server side on the etext access control. It's not deployed yet, but basically there are 2 levels of access:
- open access (that one is easy)
- fair use access (no download or viewing but search results will be available)

Now, there is a special case for the second case, which is the case where a user has special access to a fair use collection which makes it open access, but we can look at that later, it's quite rare.

Currently we cover the first type very well I think, but we need to adjust the UI a little bit for the second case:
- what happens when search results are returned in the general search? there should be an indication that the etext in not available even though search results appear
- how do we handle the search results narrowed to the collection level? (currently I think it's only possible when opening the etext viewer)
- how do we display that an etext exists on the page of an instance (for instance https://library-dev.bdrc.io/show/bdr:MW23703) but indicate that it's not available for download / display (but only search)

For the case where a user has a special access:
- it can be determined by doing a HEAD or GET request to `ldspdi/osearch/etextaccess`, which will return a 200, 401 or 403
- we should add a way to open the etext viewer for users who have access to the etext, it can be only from the instance page (and not from the general seach results)

@roopeux can you make maybe some Figma for that?

**Comment by roopeux** [comment]
<img width="732" alt="image" src="https://github.com/user-attachments/assets/ebf29daf-595a-41e5-a7ec-d1e77af22f50">

Nothing is red, nothing is a link.
Scans: Not available, Etext: not available
Add the alert icon and the text

@berger-n

**Comment by eroux** [comment]
Right, we can replace "search only" by "not available" for the etext, that's actually good!

One thing that could be very practical though would be to have the image number of the etext match... but it's a bit difficult to calculate so in a first iteration we can do without it
